### PR TITLE
Add hammer-cli-foreman-host-reports

### PR DESCRIPTION
**Pull request by ofedoren**
Supported Versions:

 * Nightly
 * 3.2

### CODE REVIEW

**Comment by zjhuntin** [comment]
[test deb]

**Comment by zjhuntin** [comment]
[test deb]

File: dependencies/bullseye/hammer_cli_foreman_host_reports/control [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+Maintainer: Debian Ruby Extras Maintainers <pkg-ruby-extras-maintainers@lists.alioth.debian.org>
+Uploaders: <>
+Build-Depends: debhelper-compat (= 12),
+               gem2deb (>= 1),
+               rake,
+               ruby-hammer-cli-foreman (<< 4.0.0),
+               ruby-hammer-cli-foreman (>= 3.0.0)
+Standards-Version: 4.5.0
+Vcs-Git: https://salsa.debian.org/ruby-team/ruby-hammer-cli-foreman-host-reports.git
+Vcs-Browser: https://salsa.debian.org/ruby-team/ruby-hammer-cli-foreman-host-reports
```
Comment (evgeni) [inline dependencies/bullseye/hammer_cli_foreman_host_reports/control:13]: this isn't true, so let's not set those options :)

File: dependencies/bullseye/hammer_cli_foreman_host_reports/control [hunk 2]
```diff
@@ -0,0 +1,25 @@
+Source: ruby-hammer-cli-foreman-host-reports
+Section: ruby
+Priority: optional
+Maintainer: Debian Ruby Extras Maintainers <pkg-ruby-extras-maintainers@lists.alioth.debian.org>
+Uploaders: <>
```
Comment (evgeni) [inline dependencies/bullseye/hammer_cli_foreman_host_reports/control:5]: this isn't true, as this package is not in debian, please set to yourself, or some generic foreman address.

File: dependencies/bullseye/hammer_cli_foreman_host_reports/salsa-ci.yml [hunk 3]
```diff
@@ -0,0 +1,4 @@
+---
```
Comment (evgeni) [inline dependencies/bullseye/hammer_cli_foreman_host_reports/salsa-ci.yml:1]: you can drop this file, it;s unused in our case

**Comment by ofedoren** [comment]
> tests fail with `File does not exist: minitest-spec-context` -- missing dependency?

Well, my assumption was that since it's required in test files only and this gem is installed with hammer-cli-foreman's Gemfile (https://github.com/theforeman/hammer-cli-foreman/blob/master/Gemfile#L11) I could simply require it in h-c-f-host-reports as well.

Please don't tell me that now I need to update Gemfile/gemspec and release a new version just because of this...

**Comment by evgeni** [comment]
> > tests fail with `File does not exist: minitest-spec-context` -- missing dependency?
> 
> Well, my assumption was that since it's required in test files only and this gem is installed with hammer-cli-foreman's Gemfile (https://github.com/theforeman/hammer-cli-foreman/blob/master/Gemfile#L11) I could simply require it in h-c-f-host-reports as well.
> 
> Please don't tell me that now I need to update Gemfile/gemspec and release a new version just because of this...

That I didn't say.

The packaging of `hammer-cli-foreman` doesn't depend on `minitest-spec-context` (and, it seems, doesn't execute any tests at all). Given `m-s-c` isn't packaged in Debian, I think the easiest would be to not execute tests here too, but I am not exactly sure what triggers them.

File: dependencies/bullseye/hammer_cli_foreman_host_reports/ruby-tests.rake [hunk 4]
```diff
@@ -0,0 +1,6 @@
+require 'gem2deb/rake/testtask'
```
Comment (evgeni) [inline dependencies/bullseye/hammer_cli_foreman_host_reports/ruby-tests.rake:1]: can you try dropping this file and see what happens? I think that's the one that executes the tests for us.

**Comment by ofedoren** [comment]
ok to test

File: dependencies/bullseye/hammer_cli_foreman_host_reports/control [hunk 5]
```diff
@@ -0,0 +1,23 @@
+Source: ruby-hammer-cli-foreman-host-reports
+Section: ruby
+Priority: optional
+Maintainer: Oleh Fedorenko <ofedoren@redhat.com>
+Uploaders: <>
+Build-Depends: debhelper-compat (= 12),
+               gem2deb (>= 1),
```
Comment (evgeni) [inline dependencies/bullseye/hammer_cli_foreman_host_reports/control:7]:
```suggestion
               gem2deb,
```

we don't need >=1 and it's not available in buster

File: dependencies/bullseye/hammer_cli_foreman_host_reports/control [hunk 6]
```diff
@@ -0,0 +1,23 @@
+Source: ruby-hammer-cli-foreman-host-reports
+Section: ruby
+Priority: optional
+Maintainer: Oleh Fedorenko <ofedoren@redhat.com>
+Uploaders: <>
```
Comment (evgeni) [inline dependencies/bullseye/hammer_cli_foreman_host_reports/control:5]:
```suggestion
```

File: dependencies/bullseye/hammer_cli_foreman_host_reports/upstream/metadata [hunk 7]
```diff
@@ -0,0 +1,9 @@
+---
+Archive: GitHub (FIXME)
```
Comment (ekohl) [inline dependencies/bullseye/hammer_cli_foreman_host_reports/upstream/metadata:2]: What's the FIXME part here?

Comment (evgeni) [reply dependencies/bullseye/hammer_cli_foreman_host_reports/upstream/metadata:2]: I have no idea what this file is, anyways ;)

Comment (ofedoren) [reply dependencies/bullseye/hammer_cli_foreman_host_reports/upstream/metadata]: https://wiki.debian.org/UpstreamMetadata#Fields doesn't tell much and I didn't find an example there :(

**Comment by ofedoren** [comment]
@evgeni, all seems to be OK now :)

**Review by evgeni — APPROVED** [review]

**Pull request merged**